1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,223 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "fileName",
+        type = "BString",
+        defaultValue = "null",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "path",
+        type = "BString",
+        defaultValue = "null",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "size",
+        type = "BNumber",
+        defaultValue = "0",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "createdDate",
+        type = "BAbsTime",
+        defaultValue = "BAbsTime.now()",
```

Comment:
You have to be careful when setting a defaultValue of BAbsTime.now(). Since Properties are static fields, BAbsTime.now() is only going to get called once for this class, not once for each instance of it (the actual call to BAbsTime.now() for the createdDate field is on line 139). So every instance of BFile that gets added to your station is going to have the same default value for createdDate. For time values, I'd recommend using BAbsTime.DEFAULT as the defaultValue like you did for modifiedDate. If you want to set the createdDate and/or modifiedDate Properties when BFile is mounted, you could always do that in the started callback.